starknet_py/utils/typed_data.py

```diff
@@ -242,58 +272,114 @@ def _verify_types(self):
             if type_name in self.types:
                 raise ValueError(f"Reserved type name: {type_name}")
 
-        referenced_types = {
-            ref_type.contains
-            if ref_type.contains is not None
-            else strip_pointer(ref_type.type)
-            for type_name in self.types
-            for ref_type in self.types[type_name]
-        }
+        referenced_types = set()
+        for type_name in self.types:
+            for ref_type in self.types[type_name]:
+                if isinstance(ref_type, MerkleTreeParameter):
+                    referenced_types.add(ref_type.contains)
+                elif isinstance(ref_type, EnumParameter):
+                    self._validate_enum_type()
+                    referenced_types.add(ref_type.contains)
+                else:
+                    referenced_types.add(strip_pointer(ref_type.type))
```

Comment:
Is it expected that we add enum variant types here like `(u128)`? I think it doesn't break anything, but it should be handled separately.

Reply:
Yes, there's actually a [test with such case](https://github.com/software-mansion/starknet.py/pull/1382/files#diff-acfd7a6c6e0206824de06e6c789bf7df469ea1261e4440a685ef2357c55058d6R13).

Reply:
Correct, I've added a check for it 👍